base/test.jl

```diff
@@ -823,4 +824,43 @@ function test_approx_eq_modphase{S<:Real,T<:Real}(
     end
 end
 
+"""
+    detect_ambiguities(mod1, mod2...)
+
+Test for ambiguous pairs of methods among the listed modules. Returns
+an array-of-`(Method,Method)` tuples.
+"""
+function detect_ambiguities(mods...; imported::Bool=false)
+    function sortdefs(m1, m2)
+        ord12 = m1.file < m2.file
+        if !ord12 && (m1.file == m2.file)
+            ord12 = m1.line < m2.line
+        end
+        ord12 ? (m1, m2) : (m2, m1)
+    end
+    ambs = Set{Tuple{Method,Method}}()
+    for mod in mods
+        for n in names(mod, true, imported)
+            try
+                f = eval(mod, n)
```

Comment:
is this possible to do without using `eval`? especially if we're considering putting this in `Pkg.test` in some form, this seems like a vector for abuse.